### PR TITLE
Update predominant color

### PR DESCRIPTION
**Pull request by strausr**

### CODE REVIEW

File: src/cloudinary-placeholder.component.ts [hunk 1]
```diff
@@ -40,16 +41,25 @@ export class CloudinaryPlaceHolder implements AfterContentChecked {
...
         {crop: 'crop', width: 10, height: 10, gravity: 'north_east'},
         {width: 'iw', height: 'ih', crop: 'fill'},
         {fetch_format: 'auto', quality: 'auto'}]
     }
+    const placeholderImageOptions = {
+      'vectorize': {effect: 'vectorize', quality: 1},
+      'pixelate': {effect: 'pixelate', quality: 1, fetch_format: 'auto'},
+      'blur': {effect: 'blur:2000', quality: 1, fetch_format: 'auto'},
+      'predominant': this.predominantTransform
+    }
```
Comment (asisayag2) [inline src/cloudinary-placeholder.component.ts]: @strausr , we said we're going to move that out of here... 
Please set as a member (instead of the _predominantTranform_) and in the above "if" condition, set the relevant `this.placeholderImageOptions.predominant`

File: src/cloudinary-image.component.spec.ts [hunk 2]
```diff
@@ -770,10 +770,10 @@ describe('CloudinaryImage', () => {
       expect(img.attributes.getNamedItem('src').value).toEqual(jasmine.stringMatching('image/upload/c_fit,w_30/e_pixelate,f_auto,q_1/bear'));
     }));
   });
-  describe('placeholder type solid', () => {
+  describe('placeholder type predominant', () => {
     @Component({
       template: `<cl-image public-id="bear" width="300" crop="fit">
-          <cl-placeholder type="solid"></cl-placeholder>
+          <cl-placeholder type="predominant"></cl-placeholder>
```
[the remark this answers is not in the document]
Comment (strausr) [reply src/cloudinary-image.component.spec.ts]: updated to predominant-color

File: src/cloudinary-image.component.spec.ts [hunk 3]
```diff
@@ -855,4 +855,67 @@ describe('CloudinaryImage', () => {
...
     }));
   });
+  describe('cl-image with acessibility modes', () => {
+    @Component({
+      template: `<cl-image public-id="bear" accessibility="darkmode"></cl-image>
+      <cl-image public-id="bear" accessibility="monochrome"></cl-image>
+      <cl-image public-id="bear" accessibility="brightmode"></cl-image>
+      <cl-image public-id="bear" accessibility="colorblind"></cl-image>`
+    })
+    class TestComponent {}
```
Comment (asisayag2) [inline src/cloudinary-image.component.spec.ts]: @strausr how is that related to this PR?

File: src/cloudinary-image.component.spec.ts [hunk 4]
```diff
@@ -770,10 +770,10 @@ describe('CloudinaryImage', () => {
       expect(img.attributes.getNamedItem('src').value).toEqual(jasmine.stringMatching('image/upload/c_fit,w_30/e_pixelate,f_auto,q_1/bear'));
     }));
   });
-  describe('placeholder type solid', () => {
+  describe('placeholder type predominant-color', () => {
```
Comment (asisayag2) [inline src/cloudinary-image.component.spec.ts]: @strausr , let's have a test for both cases (we have/don't have width/height)

File: src/cloudinary-image.component.ts [hunk 5]
```diff
@@ -121,6 +125,10 @@ export class CloudinaryImage
       if (this.placeholderComponent) {
         this.placeholderHandler(options);
       }
+      if (this.accessibility) {
+        this.options = options;
+        options.src = this.accessibilityModeHandler();
+      }
```
Comment (asisayag2) [inline src/cloudinary-image.component.ts]: @strausr , please remove

File: src/cloudinary-image.component.ts [hunk 6]
```diff
@@ -43,6 +45,7 @@ export class CloudinaryImage
 
   observer: MutationObserver;
   shouldShowPlaceHolder: boolean = true;
+  options: object = {};
```
Comment (asisayag2) [inline src/cloudinary-image.component.ts]: @strausr , please remove

File: src/cloudinary-image.component.ts [hunk 7]
```diff
@@ -141,8 +149,13 @@ export class CloudinaryImage
...
     Object.keys(options).forEach(name => {
-      placeholderOptions[name] = (name === 'width' && !options[name].startsWith('auto') || name === 'height') ? Math.floor(parseInt(options[name], 10) * 0.1) : options[name];
+      placeholderOptions[name] = (name === 'width' && !options[name].startsWith('auto') || name === 'height') ? Math.ceil(parseInt(options[name], 10) * 0.1) : options[name];
     })
     this.placeholderComponent.options = placeholderOptions;
   }
+
+  accessibilityModeHandler() {
+    return this.cloudinary.url(this.publicId, {transformation: [this.options, accessibilityEffect[this.accessibility]]});
+  }
```
Comment (asisayag2) [inline src/cloudinary-image.component.ts]: @strausr , please remove

File: src/cloudinary-placeholder.component.ts [hunk 8]
```diff
@@ -40,17 +40,10 @@ export class CloudinaryPlaceHolder implements AfterContentChecked {
...
-      'blur': {effect: 'blur:2000', quality: 1, fetch_format: 'auto'},
-      'solid': [
-        {width: 'iw_div_2', aspect_ratio: 1, crop: 'pad', background: 'auto'},
-        {crop: 'crop', width: 10, height: 10, gravity: 'north_east'},
-        {width: 'iw', height: 'ih', crop: 'fill'},
-        {fetch_format: 'auto', quality: 'auto'}]
+    if (this.type === 'predominant-color' && this.itemHeight && this.itemWidth) {
+      return this.cloudinary.url(this.publicId, {transformation: [this.options, ...predominantColorTransformPxl]});
+    } else {
+      return this.cloudinary.url(this.publicId, {transformation: [this.options, ...placeholderImageOptions[this.type] || placeholderImageOptions['blur']]})
```
Comment (asisayag2) [inline src/cloudinary-placeholder.component.ts]: @strausr MUCH better!

File: src/constants.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
+export const accessibilityEffect = {
+  'darkmode': {effect: 'tint:75:black'},
+  'brightmode': {effect: 'tint:50:white'},
+  'monochrome': {effect: 'grayscale'},
+  'colorblind': {effect: 'assist_colorblind'}
+};
```
Comment (asisayag2) [inline src/constants.ts]: @strausr, please remove
